crates/burn-train/src/metric/confusion_stats.rs

```diff
@@ -34,7 +35,7 @@ impl<B: Backend> ConfusionStats<B> {
         predictions: Tensor<B, 2>,
         targets: Tensor<B, 2, Bool>,
         threshold: Option<f64>,
-        top_k: Option<usize>,
+        top_k: Option<NonZeroUsize>,
```

Comment:
We should probably make the `top_k` for the precision metric `NonZeroUsize` as well then :)

Reply:
changed, but I think it might be easier for the user if `.with_top_k()` uses usize as a parameter no?

Reply:
`with_top_k(...)` could still take a `usize` as input but internally convert it to `NonZeroUsize`, e.g.

```rust
let k = NonZeroUsize::new(k).expect("top_k must be non-zero");
```

Reply:
ay ay, shall do that! :star: